src/app/settings/action-triggers/components/action-triggers-import/action-triggers-import.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, Input } from '@angular/core'
+import { ActionTrigger, CreateActionTrigger, KeyboardAndSelectionTriggerData, KeyboardTriggerData } from 'src/app/shared/models/action-trigger'
+import { MatSnackBar } from '@angular/material/snack-bar'
+import { ActionTriggerParser } from 'src/app/shared/abstractions/action-trigger-parser.service'
+import { ActionTriggerService } from 'src/app/shared/abstractions/action-trigger.service'
+
+@Component({
+  selector: 'sofie-action-triggers-import',
+  templateUrl: './action-triggers-import.component.html',
+  styleUrls: ['./action-triggers-import.component.scss'],
+})
+export class ActionTriggersImportComponent {
+  @Input() public actionsTriggersList: ActionTrigger<KeyboardAndSelectionTriggerData>[]
+  @Input() public disabled: boolean
+  private importedActionsTriggersList: ActionTrigger<KeyboardAndSelectionTriggerData>[]
+
+  constructor(
+    private readonly actionTriggerService: ActionTriggerService,
+    private readonly actionTriggerParser: ActionTriggerParser,
+    private readonly snackBar: MatSnackBar
+  ) {}
+
+  public fileUpload(event: Event, htmlInput: HTMLInputElement): void {
+    const files = (event.target as HTMLInputElement).files
+    if (files && files[0]) {
+      const reader = new FileReader()
+      // eslint-disable-next-line
+      reader.onload = (e: any): void => {
+        try {
+          const importedActionTriggersList: ActionTrigger<KeyboardAndSelectionTriggerData>[] = this.actionTriggerParser.parseActionTriggers(JSON.parse(e.target.result))
+          if (importedActionTriggersList.length > 0) {
+            this.importedActionsTriggersList = importedActionTriggersList
+            this.importItem(0)
+          } else {
+            this.openDangerSnackBar('No items to be added')
+          }
+        } catch {
+          this.openDangerSnackBar('Error in imported file')
+        }
+        htmlInput.value = ''
+      }
+      reader.readAsText(files[0])
+    }
+  }
+
+  private isHaveNextElement(index: number): boolean {
```

Comment:
Just a thought! 
The name doesn't relate what the method does, i'd rename it to something like `doesContainImportenActionTrigerAfterIndex` because i find it hard to achieve that. 
More ideal could be to switch the logic around and instead go with e.g.: `hasNoImportedActionTriggerAfterIndex`

Reply:
Done